gammapy/datasets/utils.py

```diff
@@ -203,3 +214,165 @@ def split_dataset(dataset, width, margin, split_template_models=True):
+
+
+    Parameters
+    ----------
+    obs : `~gammapy.data.Observation`
+        Observation containing DL4 IRFs
+    geom : `~gammapy.maps.WcsGeom`, optional
+        Output dataset maps geometry. The default is None, and it is derived from IRFS
+    energy_axis_true : `~gammapy.maps.MapAxis`, optional
+        True energy axis used for IRF maps. The default is None, and it is derived from IRFS
+    name : str, optional
+        Dataset name. The default is None, and the name is randomly generated.
+
+    Returns
+    -------
+    dataset : `~gammapy.datasets.MapDataset`
+        Map dataset.
+
+    """
+    from gammapy.makers import MapDatasetMaker
+    from gammapy.datasets import MapDataset
+
+    # define target geom
+    if geom is None:
+        geom_image = obs.aeff.geom.to_image()
+        geom = geom_image.to_cube([obs.edisp.edisp_map.geom.axes["energy"]])
+
+    energy_axis = geom.axes["energy"]
+
+    if energy_axis_true is None:
+        # energy_axis_true = obs.edisp.edisp_map.geom.axes["energy_true"]
+        energy_axis_true = obs.aeff.geom.axes["energy_true"]
+
+    # ensure that DL4 IRFs have the axes
+    rad_axis = obs.psf.psf_map.geom.axes["rad"]
+    geom_psf = obs.psf.psf_map.geom.to_image().to_cube([rad_axis, energy_axis_true])
+    geom_edisp = obs.edisp.edisp_map.geom.to_image().to_cube(
+        [energy_axis, energy_axis_true]
+    )
+    geom_exposure = geom.to_image().to_cube([energy_axis_true])
+
+    # create dataset and run data reduction / irfs interpolation
+    dataset = MapDataset.from_geoms(
+        geom,
+        geom_exposure=geom_exposure,
+        geom_psf=geom_psf,
+        geom_edisp=geom_edisp,
+        name=name,
+    )
+
+    selection = ["exposure", "edisp", "psf"]
+    if obs.events:
+        selection.append("counts")
+    if obs.bkg:
+        selection.append("background")
+
+    maker = MapDatasetMaker(selection=selection)
+    dataset = maker.run(dataset, obs)
+
+    if "background" not in selection:
+        dataset.background = Map.from_geom(geom, data=0.0)
+
+    return dataset
+
+
+def add_fermi_isotropic_model(dataset, isotropic_filepath, as_bkg=True):
+    """Add Fermi-LAT isotropic model to dataset
+
+
+    Parameters
+    ----------
+    dataset : `~gammapy.datasets.MapDataset`
+        Map dataset.
+    isotropic_filepath : str
+        Isotropic file path.
+    as_bkg : bool, optional
+        If True set the isotropic as background model instead of a SkyModel.
+        The default is True.
+    """
+
+    if dataset.models is None:
+        models = Models()
+
+    diffuse_iso = create_fermi_isotropic_diffuse_model(
+        filename=isotropic_filepath, interp_kwargs={"extrapolate": True}
+    )
+    diffuse_iso.apply_irf["edisp"] = False
+
+    dataset.models = models + [diffuse_iso]
+
+    if as_bkg:
+        dataset.background = dataset.npred()
+        dataset.models = models + [FoVBackgroundModel(dataset_name=dataset.name)]
+
+
+def create_map_dataset_from_fermi_files(
```

Comment:
See my comment above. I think this should move to `datasets/io.py` as a new `FermiPyDatasetReader`. Such that `MapDataset.read("fermipy-index.yaml")` or `MapDataset.read("path/to/fermipy/session")` (in case of fixed filenames) works.